File: frontend/src/api/content.ts
import { apiRequest } from "./client";

export type EducationalContent = {
  id: number;
  title: string;
  summary?: string | null;
  target_species?: string | null;
};

export async function listEducationalContent(): Promise<EducationalContent[]> {
  return apiRequest<EducationalContent[]>("/content/");
}

File: backend/routes/content.py
from flask import Blueprint, jsonify

from models.content import EducationalContent

content_bp = Blueprint("content", __name__)


@content_bp.get("/")
def list_contents():
    contents = EducationalContent.query.order_by(
        EducationalContent.created_at.desc()
    ).all()
    return jsonify(
        [
            {
                "id": c.id,
                "title": c.title,
                "summary": c.summary,
                "target_species": c.target_species,
            }
            for c in contents
        ]
    )
